pyomo/contrib/preprocessing/tests/test_equality_propagate.py

```diff
@@ -0,0 +1,138 @@
+"""Tests the equality set propagation module."""
+import pyutilib.th as unittest
+from pyomo.environ import (ConcreteModel, Constraint, TransformationFactory,
+                           Var, value)
+
+__author__ = "Qi Chen <https://github.com/qtothec>"
+
+
+class TestEqualityPropagate(unittest.TestCase):
```

Comment:
There are a few tests that I think need to be added:
1. A test to verify the exception throwing logic in your equality propagation transformation
2. Tests for both transformations that include more than one equality set just to be sure all the looping logic is correct (i.e. v1 == v2 and v3 == v4)
3. Tests for both transformations on indexed constraints and variables
4. Tests both transformations on models with constraints that don't fit the format you're looking for to verify that your `_build_equality_set` function will ignore them.

Reply:
I agree with Bethany's test requests.